tests/behat/features/bootstrap/Page/PageBase.php

```diff
@@ -284,36 +284,35 @@ public function waitForPage()
         }
     }
 
-    /**
-     * Wait Until Text is gone.
-     *
-     * @param int $waitmax
-     *    Number of waits of 300 ms.
-     * @param string $txt
-     *    Text that we wait for will disappear.
-     *
-     * @When waiting up to :waitmax until :txt goes away
-     *
-     * @return string
-     *    Return status of operation.
-     */
-    public function waitUntilTextIsGone($waitmax, $txt)
-    {
-        // First see if we can find the element.
-        $wait = $this->find('xpath', "//text()[contains(.,'" . $txt . "')]/..");
-        $continueWaiting = true;
-        if (!$wait) {
-            return "";
-        }
-        // Now wait for the assigned time until we no longer can find txt on the page.
-        while ($continueWaiting and --$waitmax > 0) {
-            usleep(300);
-            $wait = $this->find('xpath', "//text()[contains(.,'" . $txt . "')]/..");
-            $continueWaiting = ($wait === null);
-        }
-        if ($waitmax > 0) {
-            return "";
-        }
-        return "Failed - " . $txt . " is still on page.";
-    }
+  /**
+   * Wait Until Text is gone.
+   *
+   * @param int $waitmax
+   *    Number of waits of 300 ms.
+   * @param string $txt
+   *    Text that we wait for will disappear.
+   *
+   * @When waiting up to :waitmax until :txt goes away
+   *
+   * @return string
+   *    Return status of operation.
+   */
+  public function waitUntilTextIsGone($waitmax, $txt) {
```

Comment:
I have removed the The `waitUntilTextIsGone()` in `LibContext` as that wasn't used.

The `waitUntilTextIsGone()` in `PageBase` is called in `SearchPage` ([L595](https://github.com/ding2/ding2/blob/924b6d55a4387932c144999a9f431c76b983467c/tests/behat/features/bootstrap/Page/SearchPage.php#L595)). I don't think we can call `assertTextDoesNotAppear()` instead because i don't think we can access methods in `LibContext` from a page object but I could be wrong.

In general there are a number of methods that seems to be duplicated - if not in name then in functionality - across the "context space" and the "page object space". I'm not sure what the elegant solution is here if we want to avoid code duplication.

Reply:
:+1: